### PR TITLE
scripts: add option to dry-run

### PR DESCRIPTION
**Pull request by sebastiaanspeck**
In addition to #11810, I have added an option to perform a dry-run for both scripts. This enables the option to quickly run the script with `-s`/`--sync` without messing up your whole local repo, as you might be interested in the paths for a specific language. After this change you can run `scripts/set-more-info-link.py -Sn` & `scripts/set-alias-page.py -Sn` and even add `grep pages.{{language_code_of_interest}}` to only list the pages that would be added/updated for your language of interest.

### CODE REVIEW

**Comment by kbdharun** [comment]
@sebastiaanspeck Can you reformat the Python scripts using `black`?

File: scripts/set-alias-page.py [hunk 1]
```diff
@@ -21,13 +21,20 @@
...
     -S, --sync
         Synchronize each translation's alias page (if exists) with that of the English page.
+    -n, --dry-run
+        Show what changes would be made without actually modifying the page.
+
 
 Examples:
     1. Add 'vi' as an alias page of 'vim':
-    python3 scripts/set-alias-page.py -p common/vi vim
+       python3 scripts/set-alias-page.py -p common/vi vim
```
Comment (kbdharun) [inline scripts/set-alias-page.py:30]:
```suggestion
       python3 scripts/set-alias-page.py -p common/vi vim
       python3 scripts/set-alias-page.py --page common/vi vim
```

File: scripts/set-alias-page.py [hunk 2]
```diff
@@ -21,13 +21,20 @@
...
 
     2. Read English alias pages and synchronize them into all translations:
-    python3 scripts/set-alias-page.py -S
+       python3 scripts/set-alias-page.py -S
```
Comment (kbdharun) [inline scripts/set-alias-page.py:33]:
```suggestion
       python3 scripts/set-alias-page.py -S
       python3 scripts/set-alias-page.py --sync
```

**Review by kbdharun — COMMENTED** [review]
I have some minor suggestions for the docs, will test the commands locally now.

**Review by kbdharun — APPROVED** [review]
LGTM, tested the flags locally and everything works as intended.

File: scripts/set-alias-page.py [hunk 3]
```diff
@@ -151,45 +159,57 @@ def set_alias_page(file, command):
...
-        f.write(text)
+    if dry_run:
+        status = f"page will be {action}"
+    else:
+        status = f"page {action}"
+
+    status = f"{status_prefix}{status}\x1b[0m"
+
+    if not dry_run:  # Only write to the file during a non-dry-run
+        alias_name = os.path.basename(file[:-3])
```
Comment (blueskyson) [inline scripts/set-alias-page.py:178]: I found `file[:-3]` is used to strip `.md` extension. I think this expression can be modified for readability, for example:
```py
import os
file = "pages/linux/mkfs.ext4.md"
alias_name, _ = os.path.splitext(os.path.basename(file)) # relpace file[:-3] with splitext
print(alias_name)
```
Anyway, it depends on whether you guys think file[:-3] is fine.

**Review by blueskyson — APPROVED** [review]

**Pull request merged**